task1/task1.go

```diff
@@ -0,0 +1,74 @@
+package task1
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"strings"
+)
+
+type server struct {
+}
+
+type Query struct {
+	Key   interface{} `json:"key"`
+	Value interface{} `json:"value"`
+}
+
+type Headers struct {
+	Accept    []string `json:"Accept"`
+	UserAgent []string `json:"User-Agent"`
+}
+
+type Response struct {
+	Host       string  `json:"host"`
+	UserAgent  string  `json:"user_agent"`
+	RequestURI string  `json:"request_uri"`
+	Headers    Headers `json:"headers"`
+	Querys     []Query `json:"querys"`
+}
+
+func (s *server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	var qr []Query
+	for key, value := range r.URL.Query() {
+		var res interface{} = value
```

Comment:
In practice you'd avoid this `interface{}`, cuz you really know which params server is awaiting. Here it's OK.